realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -91,7 +91,7 @@ public void realmClosed(RealmConfiguration configuration) {
             // make sure the user is still valid
             SyncUser user = syncConfig.getUser();
             if (!user.isValid()) {
-                if (user.getAccessToken() == null) {
+                if (user.isLoggedOut()) {
```

Comment:
`RealmLog.warn("Can not use the provided configuration to open a Realm, the SyncUser is no longer valid.");` will be triggered if the SyncUser, didn't call explicitly `logout` but became invalid, mostly because of an expired `refresh_token`, for example